### PR TITLE
feat: include a legend image for categorical colorby

### PR DESCRIPTION
**Pull request by seve**
Adds an image of the left sidebar category in the case that one is selected for coloring when capturing an image.
CELLxGENE_legend (17)

### CODE REVIEW

**Comment by seve** [comment]
need to update snapshots

File: client/src/components/graph/graph.tsx [hunk 1]
```diff
@@ -1033,68 +1054,98 @@ class Graph extends React.Component<GraphProps, GraphState> {
...
-            PADDING + nodeTop,
-            nodeWidth,
-            nodeHeight
+      let categoricalLegendImageURI: string | null = null;
+
+      if (colors.colorMode !== null) {
+        if (colors.colorMode !== "color by categorical metadata") {
+          const legendNodes =
+            document.getElementById("continuous_legend")?.childNodes;
+          legendNodes?.forEach(async (node) => {
```
Comment (tihuan) [inline client/src/components/graph/graph.tsx:1067]: doesn't seem like we're awaiting, so maybe we can remove `async`?

File: client/src/components/graph/graph.tsx [hunk 2]
```diff
@@ -1033,68 +1054,98 @@ class Graph extends React.Component<GraphProps, GraphState> {
...
+                node as HTMLCanvasElement,
+                PADDING + nodeLeft,
+                PADDING + nodeTop,
+                nodeWidth,
+                nodeHeight
+              );
+            } else if (node.nodeName === "svg" || node.nodeName === "SVG") {
+              const svgNode = node as SVGSVGElement;
+              const xml = new XMLSerializer().serializeToString(svgNode);
+              const svg64 = btoa(unescape(encodeURIComponent(xml)));
```
Comment (tihuan) [inline client/src/components/graph/graph.tsx:1085]: I'm getting deprecated usage warning for `unescape` - worth checking 🙆‍♂️ ?

Screenshot 2024-04-15 at 8 02 52 AM

File: client/src/components/graph/graph.tsx [hunk 3]
```diff
@@ -1033,68 +1054,98 @@ class Graph extends React.Component<GraphProps, GraphState> {
...
-            node as HTMLCanvasElement,
-            PADDING + nodeLeft,
-            PADDING + nodeTop,
-            nodeWidth,
-            nodeHeight
+      let categoricalLegendImageURI: string | null = null;
+
+      if (colors.colorMode !== null) {
+        if (colors.colorMode !== "color by categorical metadata") {
+          const legendNodes =
```
Comment (tihuan) [inline client/src/components/graph/graph.tsx:1065]: should we extract these functions out of `graph.tsx` and import them here to use? I think it'll help us tame the readability of this giant file!

**Review by tihuan — COMMENTED** [review]
Looks really good! Just some small comments otherwise LGTM thank you ⭐ 🚀 !

**Review by kaloster — APPROVED** [review]
Well done! LGTM 🚀 

Edit: read through the code several times and it looks good to me. Submitted my review before noticing Timmy's thorough comments 🙌🏽  Agree that this file should be refactored and extract some of the logic out.

**Pull request merged**